### ISSUE TITLE
Dragging the iOS text selection handles is jumpy

### ISSUE THREAD

**Issue by justinmc** [flutter/flutter#106703]
Dragging the handles horizontally in a multi-line text field on iOS is really inaccurate and not smooth.

![out](https://user-images.githubusercontent.com/389558/176048578-fbf12e06-4826-4b6f-9615-ac4df62fc714.gif)

This definitely didn't happen before, so I've marked it as a regression.  Android doesn't seem to have the same problem.  It also doesn't seem to happen on the first line of text.

### Steps to reproduce

  1. Run any app with a multiline text field on iOS, such as the one below.
  2. Enter a few lines of text.
  3. Select some text.
  4. Drag either handle in a horizontal direction back and forth a few times

Expected: The selection and handle follow the pointer smoothly.
Actual: The selection and handle pause and jump around.

<details>

<summary>Code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            TextField(minLines: 1, maxLines: 4),
          ],
        ),
      ),
    );
  }
}
```

</details>

CC @antholeole who discovered this.

**Comment by JStevens127** [comment]
@justinmc Would it be possible for the core Flutter SDK to adopt native text inputs? Something like https://github.com/henryleunghk/flutter-native-text-input. 

Like the keyboard, there is just a never-ending (and always-changing) amount of nuance with textfields on iOS and Android devices. Flutter would instantly feel SO much smoother and more native if it relied on native inputs.

**Comment by justinmc** [comment]
That's a big question!  Generally, Flutter's approach has been to render this sort of thing internally in order to provide identical pixel perfect rendering on all platforms.  If Flutter were to pivot to always rendering native inputs, there would be a lot of work to do to support everything that the Flutter framework currently allows with its text inputs in a way that's sufficiently cross-platform.

That said, you're totally right about the native fidelity problems that we're always chasing here.  Flutter is an open source project and we're always open to better ways of doing things.  The process to make this kind of change would probably be to show that the alternate approach (like the package you linked) handles everything that the current text input stack does while eliminating these kinds of bugs, if that's possible, and to come up with a plan for integrating that approach into the framework.

In the meantime, I'll definitely keep my eyes open for alternate approaches like this.  Thanks for the kind words about Flutter and for asking the hard questions.

**Comment by TahaTesser** [comment]
Hey @justinmc 
I can reproduce this issue in Flutter, however, on native, dragging hides the text selection toolbar and shows magnification instead.

### Native when dragging handle. 

https://user-images.githubusercontent.com/48603081/182249156-3be22661-7a88-4ec3-a1b9-b36f035c1211.mov

**Comment by justinmc** [comment]
@TahaTesser You're right, but fortunately @antholeole has that part covered! https://github.com/flutter/flutter/pull/107477

I'll revisit this issue after that PR is merged, but I tried it and saw this bug in that PR too as of a week ago or so.

**Comment by iosephmagno** [comment]
@justinmc here is a concrete example using our version of the native input field wrapper. Being native it is fast and with a perfect platform fidelity. On the contrary, doing the same things shown in below video with flutter textfield is impossible.

https://user-images.githubusercontent.com/83366073/183712393-22c2ced5-bd06-45ae-9154-0cf62ee67b47.MOV

**Comment by iosephmagno** [comment]
What you can see above is:
- perfect emojis size 
- perfect toolbar and tapping behaviours
- Perfect loupe and selection
- Autocorrection
- Performance

**Comment by iosephmagno** [comment]
@justinmc I have an idea. Currently our version of plugin is ready for ios only. Android version is put up, but we had no time to fix bugs on this platform.

What if flutter team works on the plugin to make it fully functioning on android as well? And possible fine tune performance related to interfacing with the native code? Given the starting point, it would take you 2-3 weeks tops. 3 weeks to have perfect fully functioning text input on both platform seems a good deal! 🤔

This plugin would then be offered to the community as a temporary solution to all known issues while you guys finalise the text input control (whatever strategy you will decide to adopt). Just like you did with the shader warmup thing.

**Comment by iosephmagno** [comment]
@justinmc noticed the progress on single tap and loupe. May I suggest to fix emojis size? Otherwise input field would be not usable in chat and social medi apps (see pics below).

Also we noticed that on android:
- toolbar shows up on tap up after a press down, it should shows up on a press down.
- dragging selection is not working

![78E0BAEB-A0CD-4579-A4FC-A30D6996F7EF](https://user-images.githubusercontent.com/83366073/184061845-034c1b16-9077-4320-b5e8-dd765fc71cce.png)


![4375323A-D1DA-4A58-ACE6-B08DF07E2F12](https://user-images.githubusercontent.com/83366073/184061840-ea6648d2-2506-436e-9bf7-29eec26b7aa7.jpeg)

**Comment by justinmc** [comment]
@iosephmagno Thanks for bringing up the emoji issue, I'm hoping to take a closer look at what's wrong there.  I'll track it in https://github.com/flutter/flutter/issues/28894.

The native input looks good.  I don't have the bandwidth or buy-in to work on that plugin but keep it in mind in case anyone else starts thinking it would be a good idea.

**Comment by iosephmagno** [comment]
@justinmc that’s fine. Please keep me posted. We will also keep an eye on flutter textfield and check it again after you guys fix #28894. Before that, it is not an option for us

**Comment by justinmc** [comment]
I just tried this on the simulator again and it appears to be fixed with https://github.com/flutter/flutter/pull/114042.
